From bf5a9e84299a5ef0a848f2aac58cc169beb80ce8 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Wed, 18 Oct 2017 11:07:48 +0800
Subject: [PATCH 2/2] check signature with error

Cause an error rather than warning if a package only has NOKEY or
NOTTRUSTED signature.

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 lib/package.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/lib/package.c b/lib/package.c
index 1603e78..2653938 100644
--- a/lib/package.c
+++ b/lib/package.c
@@ -437,9 +437,18 @@ rpmRC rpmReadPackageFile(rpmts ts, FD_t fd, const char * fn, Header * hdrp)
 	break;
     case RPMRC_NOTTRUSTED:	/* Signature is OK, but key is not trusted. */
     case RPMRC_NOKEY:		/* Public key is unavailable. */
+
+#ifdef MANDATORY_KNOWN_SIG
+ #define MANDATORY_MSG RPMLOG_ERR
+#else
+ #define MANDATORY_MSG RPMLOG_WARNING
+#endif
 	/* XXX Print NOKEY/NOTTRUSTED warning only once. */
-    {	int lvl = (stashKeyid(keyid) ? RPMLOG_DEBUG : RPMLOG_WARNING);
+    {	int lvl = (stashKeyid(keyid) ? RPMLOG_DEBUG : MANDATORY_MSG);
 	rpmlog(lvl, "%s: %s\n", fn, msg);
+#ifdef MANDATORY_KNOWN_SIG
+	rc = RPMRC_FAIL;
+#endif
     }	break;
     case RPMRC_NOTFOUND:	/* Signature is unknown type or manifest. */
 	/* msg == NULL is probably a manifest */
-- 
1.8.3.1

